### ISSUE TITLE
indexed and sorted BAM files as Input files for "coverage" command

### ISSUE THREAD

**Issue by minsookim1983** [Ecogenomics/CheckM#71]
Dear Donovan,

Hello, this is Min-Soo Kim who is interested in your CheckM tool. I am trying to produce coverage profile from my data using "coverage" command. However, I keep getting errors with .bam files, .bai files or .sorted files as follows:
"[Error] BAM file is either unsorted or not indexed: DIRECTORY/FILE_NAME.bam"
"[Error] BAM file is either unsorted or not indexed: DIRECTORY/FILE_NAME.sorted.bai"
"[Error] BAM file is either unsorted or not indexed: DIRECTORY/FILE_NAME.sorted"

Here is the processes what I did to get sorted and indexed BAM files,

1. I used bowtie2 with paired-end read files to generate read-mapping SAM files, and converted to BAM files using samtools-v1.3.1
>samtools view -b INPUT.sam OUTPUT.bam

2. OUTPUT.bam files were sorted and indexed.
> samtools sort INPUT.bam -o OUTPUT.sorted --output_fomt BAM
> samtools index OUTPUT.sorted

3. I got .bai files, and put in OUTPUT.sorted.bai files with "coverage" command.
> checkm coverage -x fa -a 90 -t 20 BIN_FOLDER/ OUTPUT.file FILE1.sorted.bai FILE2.sorted.bai FILE3.sorted.bai ...

Could you help me figure out what is wrong with me?
Thank you for your help.

Min-Soo Kim

**Comment by dparks1134** [comment]
Hello. I assume you closed this issue as you resolved the issue. Just in case others look this up, the problem is that the `coverage` command expects the *.bam file, and not the *.bam.bai files as input.

**Comment by minsookim1983** [comment]
Thank you for your response.

As you assumed, I resolved the issue. Using samtools-v1.3.1, I did not get
"sorted.BAM file" from "BAM file". I changed to samtools-v1.2, and could
get "sorted and indexed BAM file". I found that 'coverage' command works
with "sorted BAM file" as an input file when all three files ("BAM file +
"sorted BAM file" + "sorted.bai file") are put in the same folder.

Every time I use the CheckM tool, I appreciate you who developed this
valuable and practical tool.

Min-Soo Kim
